pandasai/llm/gpt4all.py

```diff
@@ -0,0 +1,181 @@
+    we can use override certain configurations to change model's behaviour.
+
+    Example:
+        >>> import pandas as pd
+        >>> from pandasai import PandasAI
+        >>> from pandasai.llm.gpt4all import GPT4AllLLM
+
+        >>> model_name = 'ggml-replit-code-v1-3b.bin'
+        >>> additional_params = {'temp': 0, 'max_tokens': 50}
+        >>> model = GPT4AllLLM(model_name, allow_download=True, **additional_params)
+
+        >>> df_ai = PandasAI(model)
+        >>> response = df_ai(df, prompt='What is the sum of the GDP in this table?')
+
+    There is an optional parameter called model_path which sets where to
+    download the model and if it is not set then it will download the model
+    inside the folder: home/<user-name>/.local/share/nomic.ai/GPT4All/
+    Note: Please note that right now Pandas AI only supports models for gpt4all. However
+    it might not work as chatGPT when it comes to performance, hence for now users using
+    this module have to tune the existing prompts to get better results.
+    """
+
+    temp: Optional[float] = 0
+    top_p: Optional[float] = 0.1
+    top_k: Optional[int] = 40
+    n_batch: Optional[int] = 8
+    n_threads: Optional[int] = 4
+    n_predict: Optional[int] = 256
+    max_tokens: Optional[int] = 200
+    repeat_last_n: Optional[int] = 64
+    repeat_penalty: Optional[float] = 1.18
+
+    _model_repo_url = "https://gpt4all.io/models/models.json"
+    _supported_models = [
+        metadata["filename"] for metadata in requests.get(_model_repo_url).json()
+    ]
+
+    def __init__(
+        self,
+        model_name: str,
+        model_folder_path: Optional[str] = None,
+        allow_download: Optional[bool] = False,
+        n_threads: Optional[int] = 4,
+        download_chunk_size: Optional[int] = 8912,
+        **kwargs,
+    ) -> None:
+        """
+        GPT4All client for using Pandas AI
+        Args:
+            model_name: The name of the model.
+            model_folder_path: The folder inside the model weights are present
+            allow_download: If True will trigger download the specified model
+            n_threads: Number of CPU threads to be used while running the model
+            download_chunk_size: The chunk size set for downloading the model
+        """
+        self.model_name = (
+            f"{model_name}.bin" if not model_name.endswith(".bin") else model_name
+        )
+        self.chunk_size = download_chunk_size
+
+        _assert_msg = (
+            "Model not Found. Please check the list of supported models:",
+            self._supported_models,
+        )
+        assert self.model_name in self._supported_models, _assert_msg
+
+        # automatically download inside that folder
+        _default_download_path = os.path.join(
+            os.path.expanduser("~"), ".local/share/nomic.ai/GPT4All/"
+        )
+
+        self.model_folder_path = (
+            _default_download_path if model_folder_path is None else model_folder_path
```

Comment:
```suggestion
            default_download_path if model_folder_path is None else model_folder_path
```
No need for "weak internal use indicator" (see [Descriptive Naming Style](https://peps.python.org/pep-0008/#descriptive-naming-styles))